### PR TITLE
add logic in p2psession for topic cert

### PR DESCRIPTION
**Pull request by qwdarrenyin**
add logic in p2psession for topic cert

### CODE REVIEW

File: libchannelserver/ChannelRPCServer.cpp [hunk 1]
```diff
@@ -277,21 +277,21 @@ void dev::ChannelRPCServer::onClientRequest(dev::channel::ChannelSession::Ptr se
...
-        case 0x30:
-        case 0x31:
-        case 0x35:
+        case CMD_AMOP_REQUEST:
+        case CMD_AMOP_RESPONSE:
+        case CMD_AMOP_MULBROADCAST:
             onClientChannelRequest(session, message);
             break;
-        case 0x32:
+        case CMD_AMOP_TOPICREQUEST:
```
Comment (bxq2011hust) [inline libchannelserver/ChannelRPCServer.cpp]: AMOP_CLIENT_TOPICS

File: libchannelserver/ChannelRPCServer.cpp [hunk 2]
```diff
@@ -584,20 +584,11 @@ void dev::ChannelRPCServer::onClientTopicRequest(
...
-                if (topic.substr(0, topicNeedCertPrefix.size()) == topicNeedCertPrefix &&
-                    versionGt2)
-                {
-                    topics->insert(std::make_pair(topic, TopicStatus::VERIFYING_STATUS));
-                }
-                else
-                {
-                    topics->insert(std::make_pair(topic, TopicStatus::VERIFYI_SUCCESS_STATUS));
-                }
+                topics->insert(std::make_pair(topic, TopicStatus::VERIFYI_SUCCESS_STATUS));
```
Comment (bxq2011hust) [inline libchannelserver/ChannelRPCServer.cpp]: I think 50 lines is enough to modify topics' type to vector<string>, maybe you can try it?

Comment (qwdarrenyin) [reply libchannelserver/ChannelRPCServer.cpp]: I have changed it to set<string> for session and service topics.
but for lines limit.I will push it in next pr.

Comment (bxq2011hust) [reply libchannelserver/ChannelRPCServer.cpp]: > I have changed it to set for session and service topics.
> but for lines limit.I will push it in next pr.

> I have changed it to set for session and service topics.
> but for lines limit.I will push it in next pr.

Do you mean that the changes you made more than 50 lines?

File: libchannelserver/ChannelRPCServer.cpp [hunk 3]
```diff
@@ -236,7 +236,7 @@ void dev::ChannelRPCServer::blockNotify(int16_t _groupID, int64_t _blockNumber)
     }
     std::shared_ptr<dev::channel::TopicChannelMessage> message =
         std::make_shared<dev::channel::TopicChannelMessage>();
-    message->setType(0x1001);
+    message->setType(CMD_AMOP_PUSH_BLOCKNUM);
```
Comment (bxq2011hust) [inline libchannelserver/ChannelRPCServer.cpp]: Can I ask that `CMD` means what?

Comment (qwdarrenyin) [reply libchannelserver/ChannelRPCServer.cpp]: in chinese "命令字", which is used to Identify different interfaces

Comment (bxq2011hust) [reply libchannelserver/ChannelRPCServer.cpp]: I think the prefix CMD is unnecessary, the number is the type of message.

File: libdevcore/TopicInfo.h [hunk 4]
```diff
@@ -34,6 +34,22 @@ enum TopicStatus
                                  // finished with not ok result
 };
 
+enum CmdForAmop
+{
+    CMD_CHANNEL_RPC_REQUEST = 0x12,      // cmd for rpc request
```
Comment (bxq2011hust) [inline libdevcore/TopicInfo.h]: In my opinion, this is message type, is not command

File: libdevcore/TopicInfo.h [hunk 5]
```diff
@@ -42,6 +58,10 @@ class TopicItem
 
 public:
     TopicItem() : topicStatus(VERIFYING_STATUS) {}
-};
+    friend bool operator<(const TopicItem& item1, const TopicItem& item2)
```
Comment (bxq2011hust) [inline libdevcore/TopicInfo.h]:
```suggestion
    bool operator<(const TopicItem& rhs)
```

**Pull request closed without merging**